supertokens_python/recipe/emailpassword/emaildelivery/services/backward_compatibility/__init__.py

```diff
@@ -41,16 +43,20 @@ def default_create_and_send_custom_email(
     async def func(user: User, password_reset_url_with_token: str, _: Dict[str, Any]):
         if ('SUPERTOKENS_ENV' in environ) and (environ['SUPERTOKENS_ENV'] == 'testing'):
             return
+        data = {}
         try:
             data = {
                 'email': user.email,
                 'appName': app_info.app_name,
                 'passwordResetURL': password_reset_url_with_token
             }
             async with AsyncClient() as client:
```

Comment:
```suggestion
        data = {
                'email': user.email,
                'appName': app_info.app_name,
                'passwordResetURL': password_reset_url_with_token
            }
        try:
            async with AsyncClient() as client:
```

Reply:
We are passing data to `handle_httpx_client_exceptions` outside try/except block. So it needs to be defined beforehand.

Reply:
See my comment again.. i have moved it above the try block

Reply:
Fixed.